test/expectations/expectations.go

```diff
@@ -83,6 +84,18 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 	}
 }
 
+func ExpectStatus(ctx context.Context, c client.Client, obj status.Object, conditions ...status.Condition) {
```

Comment:
I suggested ExpectStatusCondition, since there may be other aspects of status we need to verify that may not map well to this function, which specifically checks statusConditions